modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -787,6 +790,13 @@ public CompletableFuture<Void> closeAsync() {
         }
 
         private void enlistPartitions(MappedFragment mappedFragment, InternalTransaction tx) {
+            // It's possible to have null txState if transaction isn't started yet.
```

Comment:
Interface InternalTransaction doesn't say that can be null. If you defenetely sure that possible seems it should be reflected for interface

Reply:
But personally I would prefer additional state instead of null value

Reply:
It is a copy from another place in InternalTableImpl.

Reply:
And what?

Reply:
can you please add mention of the created ticket for all such places meanwhile we know all of they

Reply:
I changed this code. Currently, I don't shure this ticket is required.